main/observing/condition_checker.py

```diff
@@ -212,7 +212,13 @@ def rain_check(self):
                 urllib3.exceptions.InvalidHeader, requests.exceptions.ConnectionError, requests.exceptions.Timeout,
                 requests.exceptions.HTTPError):
             return None
-        api_key = re.search(r'"SUN_V3_API_KEY":"(.+?)",', self.radar.text).group(1)
+
+        #api_key = re.search(r'"SUN_V3_API_KEY":"(.+?)",', self.radar.text).group(1)
+        api_key = re.search(r'\\"SUN_V3_API_KEY(.+?)\\":\\"(.+?)\\",', self.radar.text)
+        if api_key:
+            return None
+        else:
```

Comment:
you don't want to return None if api_key exists, because in this case you do want the rest of the function to execute
it should be 
if api_key:
   api_key = api_key.group(2)